MekHQ/src/mekhq/campaign/mission/resupplyAndCaches/StarLeagueCache.java

```diff
@@ -0,0 +1,643 @@
+                case WEIGHT_MEDIUM -> new int[]{WEIGHT_LIGHT, WEIGHT_MEDIUM, WEIGHT_MEDIUM, WEIGHT_HEAVY};
+                case WEIGHT_HEAVY -> new int[]{WEIGHT_MEDIUM, WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_HEAVY};
+                case WEIGHT_ASSAULT -> new int[]{WEIGHT_MEDIUM, WEIGHT_HEAVY, WEIGHT_ASSAULT, WEIGHT_ASSAULT};
+                default -> throw new IllegalStateException("Unexpected weight: " + weight);
+            };
+            case 2, 3 -> rollOutcome = switch (weight) {
+                case WEIGHT_LIGHT -> new int[]{WEIGHT_LIGHT, WEIGHT_LIGHT, WEIGHT_LIGHT, WEIGHT_MEDIUM};
+                case WEIGHT_MEDIUM, WEIGHT_HEAVY -> new int[]{weight, weight, weight, weight};
+                case WEIGHT_ASSAULT -> new int[]{WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_ASSAULT, WEIGHT_ASSAULT};
+                default -> throw new IllegalStateException("Unexpected weight: " + weight);
+            };
+            case 4, 5 -> rollOutcome = switch (weight) {
+                case WEIGHT_LIGHT -> new int[]{WEIGHT_LIGHT, WEIGHT_LIGHT, WEIGHT_MEDIUM, WEIGHT_MEDIUM};
+                case WEIGHT_MEDIUM -> new int[]{WEIGHT_MEDIUM, WEIGHT_MEDIUM, WEIGHT_MEDIUM, WEIGHT_HEAVY};
+                case WEIGHT_HEAVY -> new int[]{WEIGHT_MEDIUM, WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_ASSAULT};
+                case WEIGHT_ASSAULT -> new int[]{WEIGHT_HEAVY, WEIGHT_ASSAULT, WEIGHT_ASSAULT, WEIGHT_ASSAULT};
+                default -> throw new IllegalStateException("Unexpected weight: " + weight);
+            };
+            case 6 -> rollOutcome = switch (weight) {
+                case WEIGHT_LIGHT -> new int[]{WEIGHT_LIGHT, WEIGHT_LIGHT, WEIGHT_MEDIUM, WEIGHT_HEAVY};
+                case WEIGHT_MEDIUM -> new int[]{WEIGHT_MEDIUM, WEIGHT_MEDIUM, WEIGHT_HEAVY, WEIGHT_HEAVY};
+                case WEIGHT_HEAVY -> new int[]{WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_ASSAULT};
+                case WEIGHT_ASSAULT -> new int[]{WEIGHT_ASSAULT, WEIGHT_ASSAULT, WEIGHT_ASSAULT, WEIGHT_ASSAULT};
+                default -> throw new IllegalStateException("Unexpected weight: " + weight);
+            };
+            default -> throw new IllegalStateException("Unexpected value in getlanceWeights(): " + roll);
+        }
+
+        for (int outcome : rollOutcome) {
+            unitWeights.add(outcome);
+        }
+
+        return unitWeights;
+    }
+
+    public void createDudDialog(StratconTrackState track, StratconScenario scenario) {
+        StratconCoords stratconCoords = scenario.getCoords();
+
+        // Dialog dimensions and representative
+        final int DIALOG_WIDTH = 400;
+        final int DIALOG_HEIGHT = 200;
+
+        // Creates and sets up the dialog
+        JDialog dialog = new JDialog();
+        dialog.setTitle(resources.getString("dialog.title"));
+        dialog.setLayout(new BorderLayout());
+        dialog.setSize(UIUtil.scaleForGUI(DIALOG_WIDTH, DIALOG_HEIGHT));
+        dialog.setLocationRelativeTo(null);
```

Comment:
If the GUIs and dialogs are just to allow users to interact with a very small a mount of things that can be interacted also through the API of the class I am more prone to be ok with it, but I still would prefer it to be a separated class which this one can instantiate and show to the user.